src/api/firebase.js

```diff
@@ -69,12 +75,15 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 	return docRef;
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listId, documentId, totalPurchases) {
+	const itemRef = doc(db, listId, documentId);
+
+	const updatedDocRef = await updateDoc(itemRef, {
+		dateLastPurchased: serverTimestamp(),
```

Comment:
any specific reason why are we using the `serverTimestamp`? I noticed that it causes a small delay before the checkbox gets checked because it waits until the server receives the document.
I would recommend using  `new Date()` and it will work without that delay.